src/main/fc/rc.c

```diff
@@ -148,6 +148,24 @@ float applyRaceFlightRates(const int axis, float rcCommandf, const float rcComma
     return angleRate;
 }
 
+float applyKissRates(const int axis, float rcCommandf, const float rcCommandfAbs)
+{
+    float kissRpyUseRates;
+    const float rcCurvef = currentControlRateProfile->rcExpo[axis] / 100.0f;
+
+    if (currentControlRateProfile->rates[axis] < 100) {
```

Comment:
This presents a difficult situation. Having the calculations only apply where the rc_rate < 100 means the rates will only be adjustable through less than half of the possible range (value is allowed to be 0-255). Even worse, the default values for the rc_rates for each axis is 100 so a user switching to KISS rates typically would by default have +-2000dps rates! It would seem more appropriate to apply some scaling factor so that rc_rate value of 100 produces a setpoint rate that's approvimately equal to the the setpoint produced by Betaflight rates. It may not be possibly to replicate the explicit rate numbers used in KISS rates.

Reply:
Maybe is better to change the value to 99 in the validation of parameters of kiss is selected

Reply:
CLI parameter validation can't work that way. The range validation can't be changed based on context of another setting. Could be done in the Configurator (kind of) but no in CLI, CMS and maybe not Lua (easily).

Reply:
@etracer65 wait, "rates" in kiss are super rates in betaflight, so default betafight rates are fine (1.0, 0.7) it produces about 666 deg /sec

Reply:
I was talking about something like you did with the filters